### ISSUE TITLE
False locale in category api return

### ISSUE THREAD

**Issue by alexander-schranz** [sulu/sulu#3643]
| Q | A
| --- | ---
| Bug? | no
| New Feature? | no
| Sulu Version | Specific version or SHA of a commit
| Browser Version | Browser name and version

#### Actual Behavior

When loading a category in a none exist locale both values defaultLocale and locale are the same.

#### Expected Behavior

Locale should be the requested locale (not sure?) and defaultLocale should be exist locale.

#### Steps to Reproduce

1. Create Category in EN
2. Switch language to DE Response will be:

```
{
  "locale": "en",
  "defaultLocale": "en"
}
```

#### Possible Solutions

Set the correct locale to the apiCategory object.

**Comment by danrot** [comment]
Since we have introduced a `ghostLocale` property in #4940 we might be able to just delete the `defaultLocale`, and set the `locale` to `de` in your example.

**Comment by alexander-schranz** [comment]
@danrot sounds good so for BC we could should deprecate the defaultLocale?

**Comment by danrot** [comment]
Yeah, definitely... But just changing the value of the `locale` field sounds like a BC break, but I guess the current behavior could be classified as bug.
